### DISCUSSION TITLE
几年了，我的移植大业仍然停留在第一步（

### DISCUSSION THREAD

**Discussion by Lavaver** [Meloong-Git/PCL#3805] [讨论]
对于只学过 C# / Java / Python 的我来说，这简直是个天书（大悲）
本来想要把 PCL2 的认证&启动过程 Fork 成类库形式（隔壁 KMCCC 库几年了都不见上微软 auth 和模组客户端支持所以就打算两者结合起来 Fork2Lib 成 KMPageLaunchLeft Lib（KMPLL））
我本以为语法上会很接近至少能让我读懂啥意思，结果开幕直接死透了：

```vb
﻿Imports System.Windows.Navigation

Public Class FormLoginOAuth

    Public Event OnLoginSuccess(code As String)
    Public Event OnLoginCanceled(IsSwitch As Boolean)
'后面的就不放了（）
```
（再次陷入思考）（思考人生存在的意义）
查了很多资料问了 GPT 才知道
丫的 `C#` 要用 `event` 关键字+委托才能不报错啊（
也就是说要重写这段天书我需要为每个需求挂委托
~凯瑟琳：向着星辰与深渊...感谢你完成了这次的委托，这是给你的奖励（10,000 错误）~

```csharp
namespace KMPLL
{
    public class FormLoginOAuth
    {
        public event Action<string> OnLoginSuccess;
        public event Action<bool> OnLoginCanceled;
    }
}
```

别急，我在思考（）

**Comment by MoYuan-CN** [comment]
Microsoft 认证过程我有 Python 类库，传那个登录结束 url 拿 access_token 的
想要参考参考我可以发出来

**Reply by Lavaver** [reply]
你知道最令我疑惑的点是什么吗——整个 vb 代码中没看到花括号的影子（事实上也确实不存在（？）花括号的概念），我真想让这个世界上每个类每个关键循环都打上`{}`（（（

**Reply by MoYuan-CN** [reply]
是这样的，只有 `XXX` 和 `End XXX`
~~我也一样喜欢 `{}` 然后吐槽 Python 的可读性~~

**Reply by MoYuan-CN** [reply]
Python MC微软登录类库：[Yuns-Lab/MCMsLogin-Python](https://github.com/Yuns-Lab/MCMsLogin-Python)

**Reply by wuliaodexiaoluo** [reply]
> 你知道最令我疑惑的点是什么吗——整个 vb 代码中没看到花括号的影子（事实上也确实不存在（？）花括号的概念），我真想让这个世界上每个类每个关键循环都打上`{}`（（（

别，太麻烦

**Reply by Lavaver** [reply]
@JingHai-Lingyun 我此前稍微接触过一点点 py ，如果可能的话会试着转一下代码
mc 启动已经有一个同样为 C# 打底的类库 [KMCCC](https://github.com/MineStudio/KMCCC) （顺带说一下 KMCCC 已经停止维护更新了，Release 也停留在了 2019 ，不过拿来当范本应该能省下不少时间吧...？），可以试着做一下 example 范本

**Reply by Lavaver** [reply]
Mod 端启动的话可能有点悬...

**Reply by Lavaver** [reply]
@JingHai-Lingyun ，https://github.com/Lavaver/KMCSharp-Framework

**Comment by Silverteal** [comment]
可能直接对着协议栈写会更方便？

**Reply by Lavaver** [reply]
这么做我迟早得似（

**Comment by Lavaver** [comment]
好消息好消息！
我终于搞完了微软正版验证器！
OHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH
接下来就是要接入一些关键部分和调试了

**Reply by Lavaver** [reply]
接下来就是搞其他的事务了
